spyder/api/plugins/new_api.py

```diff
@@ -750,11 +750,12 @@ def get_name():
 
         Notes
         -----
-        This is a method to be able to update localization without a restart.
+        This method needs to be decorated with `staticmethod`.
         """
         raise NotImplementedError('A plugin name must be defined!')
 
-    def get_description(self):
+    @staticmethod
+    def get_description():
```

Comment:
Since this changes base methods (`get_description` and `get_icon`) for the plugins seems like the API version should be updated?:

https://github.com/spyder-ide/spyder/blob/c2ce80288f8fe6c473e54bfc99a54750706ba147/spyder/api/_version.py#L7-L26

Also, since this changes de API maybe could be worthy to also eventualy update the related plugin resources (docs, example repo, plugins repos, etc) so maybe creating issues in the respective repos could be worthy to keep track of this change?

Reply:
> Since this changes base methods (get_description and get_icon) for the plugins seems like the API version should be updated?

Yep, you're right. I'll push a commit to bump our API version.

> Also, since this changes de API maybe could be worthy to also eventualy update the related plugin resources (docs, example repo, plugins repos, etc) so maybe creating issues in the respective repos could be worthy to keep track of this change?

I think the best we can do for now is to document the API changes in our changelog (I'll also add an entry there), so that plugin authors and ourselves remember what's been added/removed in version 6. Then, when we finally release it, we could do what you propose (I think right now it's premature to do that because more API could be necessary).